src/main/java/com/unorganized/dao/TaskListDAO.java

```diff
@@ -6,21 +6,21 @@
 import java.io.IOException;
 import java.io.ObjectInputStream;
 import java.io.ObjectOutputStream;
-import java.io.Serializable;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
 import com.unorganized.dto.TaskDTO;
 
-public class TaskListDAO implements Serializable
+public class TaskListDAO implements ITaskListDAO
 {	
 	private static Path filePath = Paths.get(System.getProperty("user.dir") + "\\src\\main\\resources\\taskList.ser");	
-	private static List<TaskDTO> tasks;	
+	private List<TaskDTO> tasks;
```

Comment:
These DAO methods should not be static - DAO should be instantiated before the methods can be called.